core/src/package.rs

```diff
@@ -156,3 +234,28 @@ impl Package {
         })
     }
 }
+
+pub(crate) trait ValueExt {
```

Comment:
Yeah, I was thinking of having a custom type. That would probably make more sense.

However, `serde_json::Value` contains much more than just those types we accept. If we change to a custom type, I think it is even better to make our own `enum impl Into<serde_json::Value>`. Do you agree with making it an enum instead? But I do agree that extensions aren't the nicest of things!

Reply:
I did a new enum ArgValue that has one case for each argument type we support. This will allow us to later on add enums as types and such. Note that the only actual use of "ArgValue" is to use for native packages, in all other cases ArgValue is just an intermediate step where the conversion to/from it is doing the verification, but the value is never stored anywhere. This is because that we can't use ArgValue for Element nor JsonEntry since it doesn't know of the type of the argument when .into()-ing to an Element, and we don't know the type when deserializing JsonEntry. We could split it up into two different JsonEntry types - one for serializing and one for deserializing - but I thought that was kind of complicated...